packages/s2-core/src/facet/base-facet.ts

```diff
@@ -714,6 +701,11 @@ export abstract class BaseFacet {
     this.viewCellHeights = this.getViewCellHeights();
   };
 
+  /**
+   * 提供给明细表做 rowOffsets 计算的 hook
+   */
+  protected calculateRowOffsets() {}
```

Comment:
```suggestion
  abstract calculateRowOffsets(): void
```

声明成抽样吧, 透视表那边声明 `calculateRowOffsets() {}`, 不然有点怪